driver/config/config.go

```diff
@@ -164,8 +164,9 @@ type (
 	}
 	//fandom-end
 	SelfServiceHook struct {
-		Name   string          `json:"hook"`
-		Config json.RawMessage `json:"config"`
+		Name             string          `json:"hook"`
+		PersistencePhase []string        `json:"persistence_phase"`
```

Comment:
1. Shouldn't we update some documentation to explain the purpose of this field and allowed values?
2. Would it make sense to use an enum type for it and write a custom `UnmarshalJSON` to validate phase values?

Reply:
@mkalinowski-fandom https://github.com/Wikia/kratos/blob/master/docs/docs/self-service/hooks.mdx that should help

Reply:
Additionally you should update the schema since it will not allow for this extra field at the moment. See https://github.com/Wikia/kratos/blob/master/embedx/config.schema.json#L156-L219.

Reply:
just an idea - we could probably simplify it and use a string instead of an array of strings. one could pass `pre-persist` or `post-persist` here. empty value would mean both phases. or we could allow something like the "all" value if we want it to be explicit.
getting rid of the array would allow us to simplify the code below and remove the `contains` array.

Reply:
I like the idea. Changed the code that way.

Reply:
Docs and schema updated